### ISSUE TITLE
Feature request: search cloud services

### ISSUE THREAD

**Issue by lexoyo** [silexlabs/unifile#162]
[Following this discussion in CE2 repo](https://github.com/silexlabs/CloudExplorer2/issues/88), I create an issue here for a new unifile feature

As services like [dropbox offer an API to search files and folders](https://www.dropbox.com/developers/documentation/http/documentation#files-search) (names and content and metadata), I would like to add a search box in CE2 for such services (not for the others like FTP)

You can see where I am going from there @JbIPS, could you give it a thought and we could discuss the implementation here?

**Comment by JbIPS** [comment]
You know my position on API that are not replicated on **all** services :blush:

But GitHub must have something like that and `fs` has `find`. For other, we could have a more manual approach with warnings about the slowness of the operation.

**Comment by lexoyo** [comment]
Yes why not... I guess it is not only the time it takes but also we can not find content of files with ftp for example

**Comment by lexoyo** [comment]
I guess another solution would be to create a separate project just for search? How would it share auth with unifile?

**Comment by JbIPS** [comment]
I forgot about content and metadata. That's a problem...

Does it have to do that? I mean it's easier to limit the search on Dropbox than to do it in FTP.

And regarding another project, the only solution would be to clone unifile-dropbox and use it directly without the Unifile layer. But I'm not very fond of this solution :confused:

**Comment by JbIPS** [comment]
And to be honest, we _could_ do it in FTP, by getting a file content and scanning it. That would be huge and heavy, but doable.
